aten/src/ATen/native/transformers/cuda/sdp_utils.cpp

```diff
@@ -109,6 +109,10 @@ int64_t minimum_gemm_alignment(sdp_params const& params) {
   return matmul_alignment_mn;
 }
 
+// On ROCM, ME and FA share the backend, and hence they share the checking
+// function for fundamental limitations by the GPU kernel
+// caller_is_meff is added to make the TORCH_WARN message showing the correct result
+template<bool caller_is_meff = false>
```

Comment:
@malfet to look at this change especially, since this is a function commonly used by CUDA and ROCm, but the changes here maintains the same functionality for CUDA.